### PR TITLE
Feature/dlit 1086/tool image frontend

### PR DESCRIPTION
**Pull request by helgigretar**
Frontend for tool image. The backend is also finished.

### CODE REVIEW

File: src/Pages/Tools/ToolStock/ToolsOnStock.tsx [hunk 1]
```diff
@@ -28,10 +44,12 @@ function ToolsOnStock() {
   const [allowed_days_dialog_open, setAllowedDaysDialogOpen] = useState(false);
   const [selected_row, setSelectedRow] = useState({} as ToolData);
   const [broken_lost_dialog_open, setBrokenLostDialogOpen] = useState(false);
-  const [selected_broken_lost,setSelectedBrokeLost] = useState("")
-  const [broken_lost_reason,setBrokenLostReason] = useState("");
+  const [imageDialogOpen, setImageDialogOpen] = useState(false);
+  const [selected_broken_lost, setSelectedBrokeLost] = useState("")
+  const [broken_lost_reason, setBrokenLostReason] = useState("");
```
Comment (hafthorarinsson) [inline src/Pages/Tools/ToolStock/ToolsOnStock.tsx:49]: change to camelCase?

File: src/Pages/Tools/ToolStock/ToolsOnStock.tsx [hunk 2]
```diff
@@ -42,15 +60,33 @@ function ToolsOnStock() {
...
         <TableCell>{row.shelf_location}</TableCell>
         <TableCell>{row.user}</TableCell>
+        <TableCell className={classes.toolImageContainer}>
+          {
+            row.tool_image === null ?
+              <input type="file" onChange={(e) => getBase64(row.part_number, e)} />
+              :
+              <img onClick={() => { setSelectedRow(row); setImageDialogOpen(true) }} src={row.tool_image} className={classes.toolImage} alt="multiToolImage" />
+          }
+
```
Comment (hafthorarinsson) [inline src/Pages/Tools/ToolStock/ToolsOnStock.tsx:70]: FYI. If we would want to add option to upload from camera we would just need to have accept="image/*" property on the input tag

File: src/Pages/Tools/ToolStock/ToolsOnStock.tsx [hunk 3]
```diff
@@ -85,46 +117,51 @@ function ToolsOnStock() {
   const closeAllowedDaysDialog = (status: boolean) => {
     if (status === true) {
       selected_row.days_allowed = parseInt(days_allowed);
-      toolsApi.editDaysAllowed(selected_row).then(res => {
-        toolsApi.getAllToolsFromMxi(order_by_text, ascending).then(res => {
-          dispatch({ type: 'ON_MXI_TOOLS', tools: res })
-          all_tools = res;
-          setFilteredTools(res)
-        })
+      ToolsApi.editDaysAllowed(selected_row).then(res => {
```
Comment (hafthorarinsson) [inline src/Pages/Tools/ToolStock/ToolsOnStock.tsx:120]: camelCase?

File: src/Pages/Tools/ToolStock/ToolsOnStock.tsx [hunk 4]
```diff
@@ -85,46 +117,51 @@ function ToolsOnStock() {
...
-  const onSelectedBrokenLost = (value:string)=>{
+  const onSelectedBrokenLost = (value: string) => {
     setFilteredTools([])
     setSelectedBrokeLost(value)
   }
+  const refresh = (orderByText:string, ascending:boolean)=>{
+    ToolsApi.getAllToolsFromMxi(orderByText, ascending).then(res => {
+      setImageDialogOpen(false);
+      dispatch({ type: 'ON_MXI_TOOLS', tools: res })
+      all_tools = res;
```
Comment (hafthorarinsson) [inline src/Pages/Tools/ToolStock/ToolsOnStock.tsx:150]: camelCase?

File: src/Pages/Tools/ToolStock/ToolsOnStock.tsx [hunk 5]
```diff
@@ -142,15 +179,35 @@ function ToolsOnStock() {
...
+        visible={imageDialogOpen}
+        max_width={"xl"} title={selected_row.description}
+        context={
+          <Grid className={classes.fullScreenContainer}>
+            <Typography text={"Change picture"} align="center" variant="h5" />
+            <input type="file" id="toolImage" onChange={(e) => getBase64(selected_row.part_number, e)} />
+            <br/>
+            <br/>
+            <Grid >
+              <img src={selected_row.tool_image!} className={classes.fullScreenImage} alt="user_image" />
```
Comment (hafthorarinsson) [inline src/Pages/Tools/ToolStock/ToolsOnStock.tsx:202]: Why the underscore in the alt text?
Shouldn't we just have User Image?

**Review by hafthorarinsson — APPROVED** [review]
Approved, just some minor comments :)

File: src/Pages/Tools/ToolStock/ToolsOnStock.tsx (hunk 2 above)
Comment (helgigretar) [reply src/Pages/Tools/ToolStock/ToolsOnStock.tsx:70]: Okay good to know. I will ask

File: src/Pages/Tools/ToolStock/ToolsOnStock.tsx (hunk 4 above)
Comment (helgigretar) [reply src/Pages/Tools/ToolStock/ToolsOnStock.tsx:150]: if you are talking about the type I feel like that should be snake case but if you are pointing at all_tools I agree with you and I have changed that variable name.

File: src/Pages/Tools/ToolStock/ToolsOnStock.tsx (hunk 5 above)
Comment (helgigretar) [reply src/Pages/Tools/ToolStock/ToolsOnStock.tsx:202]: yes that makes sense.

**Pull request merged**